DataFormats/EgammaCandidates/interface/GsfElectron.h

```diff
@@ -183,11 +183,18 @@ class GsfElectron : public RecoCandidate
     // forward core methods
     SuperClusterRef superCluster() const override { return core()->superCluster() ; }
     GsfTrackRef gsfTrack() const override { return core()->gsfTrack() ; }
-    virtual TrackRef closestTrack() const { return core()->ctfTrack() ; }
+    //so it turns out closestCtfTrackRef is overriden by pat::Electron but not closestTrack()
+    //this results in surprising behaviour, ie two functions which looked identical 
+    //would give different results so now closestTrack just calls closestCtfTrackRef
+    virtual TrackRef closestTrack() const { return closestCtfTrackRef() ;}
```

Comment:
[responding/continuing from https://github.com/cms-sw/cmssw/pull/26389#pullrequestreview-228411055]
It's curious that the closestTrack method was added for purposes of PAT embedding https://github.com/cms-sw/cmssw/commit/ba98a5f008648802611e5b2630137ccf2d9f52a6
However, just a few days later, the PAT side decided to use the other method
https://github.com/cms-sw/cmssw/commit/fa396c23edbf071da765c823dccb8d5990102333

git grep in the baseline shows only one place where this method is used
Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.cc
So, apparently there was a known secret not to use closestTrack  until you stepped into it in https://github.com/cms-sw/cmssw/commit/13fb21b22c6c5130721ecd37df495771cad1dcbb

Overall, this looks to me like a good case to remove this method.

Reply:
yes, this is a mess. However I prefer not to remove methods and make code not compile in 10_6 that did in 10_2, given that we want to properly fix this in 11_X